src/plugins/onePingPerUser/index.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { ChannelStore, ReadStateStore } from "@webpack/common";
+import { Message } from "discord-types/general";
+
+enum ChannelType {
+    DM = 1,
+    GROUP_DM = 3
+}
+
+export default definePlugin({
+    name: "OnePingPerUser",
+    description: "Limits the ping audio notification from playing more than once per user until messages are marked as read to prevent audio spam and hearing multiple notifications as a cue that more than one user may have messaged you",
+    authors: [Devs.ProffDea],
+    patches: [{
+        find: ".getDesktopType()===",
+        replacement: [{
+            match: /if\((\i\.\i\.getDesktopType\(\)===\i\.\i\.NEVER)\){/,
+            replace: "if($1&&$self.isPrivateChannelRead(arguments[0]?.message)){"
```

Comment:
should this not be this?
```suggestion
            replace: "if($1||$self.isPrivateChannelRead(arguments[0]?.message)){"
```

Reply:
The && and || didn't have the intended effect, the new commit fixes the issue of desktop notifications firing when they shouldn't.

Reply:
okay, now it makes more sense